File: SE_PROJECT - FINAL/Registration/Backend/models/Voter.js
// models/Voter.js
import { Schema, model } from 'mongoose';

const voterSchema = new Schema(
  {
    userId: {
      type: Schema.Types.ObjectId,
      ref: 'User',
      required: true,
      unique: true,
    },
    identityProofHash: {
      type: String,
      required: true,
      unique: true,
    },
    status: {
      type: String,
      enum: ['PENDING', 'APPROVED', 'REJECTED'],
      default: 'PENDING',
    },
  },
  {
    timestamps: true,
  }
);

const Voter = model('Voter', voterSchema);

export default Voter;

File: SE_PROJECT - FINAL/Backend/src/main/java/com/evoting/entity/Voter.java
package com.evoting.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Entity
@Table(name = "voters")
public class Voter {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "user_id", nullable = false, unique = true)
    private String userId;

    @Column(name = "identity_hash", nullable = false, unique = true)
    private String identityHash;

    // Storing encrypted identity data - "NEVER store plaintext"
    @Column(name = "encrypted_identity", nullable = false, columnDefinition = "TEXT")
    private String encryptedIdentity;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Status status = Status.PENDING;

    @Column(name = "identity_type", nullable = false)
    private String identityType;

    @CreationTimestamp
    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    public enum Status {
        PENDING,
        APPROVED,
        REJECTED
    }
}
